usr/bin/irdata_text_index.py

```diff
@@ -24,47 +24,35 @@
 with this program.  If not, see <http://www.gnu.org/licenses/>.
 """
 
-from irdata.data import RawImportFile, rewrite, index_for_int
-from os.path import split, splitext
-import os, sys, cmdsyntax, gzip
-
-syntax_description = """
-    --help |
-    (
-      <interval> [ <filename> | - ] [ -f <field> ] [ -d <separator> ]
-    )
-    """
+import argparse
+import gzip
+import os
+import sys
+from irdata import data
+
 
 if __name__ == "__main__":
     progname = os.path.basename(sys.argv[0])
 
     # Get the command line options.
+    argparser = argparse.ArgumentParser(
+        description=__doc__, formatter_class=argparse.RawDescriptionHelpFormatter
+    )
+    argparser.add_argument("-f", metavar="field-index", type=int)
+    argparser.add_argument("-d", metavar="separator", default="\t")
```

Comment:
Later, you use `args.d  # separator`. It would be nice not to have to comment this.

```suggestion
    argparser.add_argument("-d", metavar="separator", default="\t", dest="separator")
```

Reply:
Excellent suggestion, I should have done this from the start.

Reply:
:) No worries. That's what several eyes are for ;)